### DISCUSSION TITLE
set: null vs unset: true

### DISCUSSION THREAD

**Discussion by ZehuaZhang** [prisma/prisma#18946] [Q&A]
I've read the documentation on Guide and API Reference, still not clear on what's the difference between `set: null` and `unset: true`. Especially not clear what's the actual meaning of `remove the list entirely`. Does it mean to remove the elements in the list entirely which becomes an empty array. Or does it mean to remove the elements, and array itself, but that becomes a `null`, which would be the same as `setL:null`? So I'm guessing first assumption is correct?

Say:
```prisma
array String[]
```

My understanding is that 
`set: null` is `array = null`
`unset: true` is `array = []`

**Comment by nurul3101** [comment]
Hello @ZehuaZhang 👋 

> Especially not clear what's the actual meaning of remove the list entirely. Does it mean to remove the elements in the list entirely which becomes an empty array.

This would mean that the property array will be removed from the document by using `unset:true` i.e. the list will be removed(deleted) while setting `set:null` would mean that the property array exists in the document but has a null value.

**Reply by ZehuaZhang** [reply]
It's still not clear, I'm sorry. Could you help me explain based on my example?

**Reply by nurul3101** [reply]
Sure! Based on your example, if you would perform `set: null` then the value of array will be null as you expected.
So, a sample document would look like this:
```js
  {
    _id:"5f9d5f3b9d9d1a0008f4e4a1",
    array: null
  }
``` 
But if you would perfomr `unset:true` then the array field will be removed from the document, so a sample document would look like:
```js
  {
    _id:"5f9d5f3b9d9d1a0008f4e4a1"
  }
```
Notice that the array field is removed from the document. Hope this helps! Please let me know in case you have any queries.

**Reply by ZehuaZhang** [reply]
This makes more sense now, but I'm assuming you're talking about Mongo DB document? 
What about the SQL like postgres, the property or the column `array` would always be there, what would be the value for each scenario between `set: null` and `unset: true` in the table itself and in the code return from prisma crud operations?

**Reply by nurul3101** [reply]
Yes, I am talking about MongoDB here.
unset property is only supported for MongoDB and not for SQL databases - [Reference](https://www.prisma.io/docs/reference/api-reference/prisma-client-reference#unset).

**Reply by nurul3101** [reply]
If this answers your question, it would be great if you could **mark this Discussion as answered** to indicate that it has been resolved. 

Otherwise please let us know how else we can help you further or close the Discussion if it was resolved in some other way 🙏

**Reply by nurul3101** [reply]
@ZehuaZhang Do let us know in case you need any help 🙂
